### PR TITLE
Irene Ryan pull request for riddler castle datasets

### PR DESCRIPTION
**Pull request by ireneryan**

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -22,6 +22,7 @@ Authors@R: c(
     person("Jayla", "Nakayama", role = "ctb"),
     person("Natalia", "Iannucci", email = "iannuccinatalia@gmail.com", role = "ctb"),
     person("Marium", "Tapal", email = "mariumtapal@gmail.com", role = "ctb"),
+    person("Irene", "Ryan", email = "ryan.irene3@gmail.com", role = "ctb"))
```
Comment (beanumber) [inline DESCRIPTION]: You've got a trailing parenthesis here that needs to be replaced with a comma

File: process_data_set_irene.R [hunk 2]
```diff
@@ -0,0 +1,73 @@
+library(tidyverse)
```
Comment (beanumber) [inline process_data_set_irene.R]: This file needs to be in `data-raw/`, not at the root level of the repository.

File: process_data_set_irene.R [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+library(stringr)
+library(lubridate)
+library(janitor)
+library(usethis)
+
+castle_solutions <- read_csv("data-raw/riddler-castles/castle-solutions.csv")
+
+castle_solutions<-castle_solutions%>% #clean the names
```
Comment (beanumber) [inline process_data_set_irene.R]: Please put spaces around `<-` and `%>%` throughout this code.

File: process_data_set_irene.R [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  clean_names()
+colnames(castle_solutions)[11] <- "strategy"
```
Comment (beanumber) [inline process_data_set_irene.R]: Can you replace the hard-coded index here with a call to `dplyr::rename()`?

File: process_data_set_irene.R [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+castle_solutions$user <- seq.int(nrow(castle_solutions)) #create user column
+  
+#gather the data to tidy form
+castle_solutions<-gather(castle_solutions, key = "castle", value = "number_of_soldiers", castle_1, castle_2, castle_3, castle_4, castle_5, castle_6,
```
Comment (beanumber) [inline process_data_set_irene.R]: Please add carriage returns so that this doesn't go so far on one line

Comment (beanumber) [reply process_data_set_irene.R]: You could also the select helper `contains("castle_")` make this a little shorter.

File: process_data_set_irene.R [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+castle_solutions_2<-castle_solutions_2%>%
+  mutate(user = as.character(user))
+
+#-------------------------
+
+castle_solutions_3 <- read_csv("data-raw/riddler-castles/castle-solutions-3.csv")
+
+castle_solutions_3<-castle_solutions_3%>% #clean the names
+  clean_names()
+colnames(castle_solutions_3)[11] <- "strategy"
```
Comment (beanumber) [inline process_data_set_irene.R]: Ditto the previous comment on hard-coding indices.

**Review by beanumber — CHANGES_REQUESTED** [review]

File: data-raw/process_data_set_irene.R [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  clean_names()
+
+castle_solutions$user <- seq.int(nrow(castle_solutions)) #create user column
+  
+#gather the data to tidy form
+castle_solutions <- gather(castle_solutions, key = "castle", value = "number_of_soldiers", 
+                           contains("castle_"), -strategy, -user)
+
+#reorder
+castle_solutions <- castle_solutions[c("user", "castle", "number_of_soldiers", "strategy")]
```
Comment (beanumber) [inline data-raw/process_data_set_irene.R]: As per @rudeboybert 's preferences #61 , please use a `tidyverse` style. So here you would replace the `[` with a call to `select()`.

File: data-raw/process_data_set_irene.R [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+#classes of variables: only adjusted the user column, the rest are the appropriate class
+castle_solutions <- castle_solutions%>%
```
Comment (beanumber) [inline data-raw/process_data_set_irene.R]: spaces before the `%>%` please! You can use `lintr` to fix these automatically if you get sick of doing it manually!

File: data-raw/process_data_set_irene.R [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+
+castle_solutions_2$user <- seq.int(nrow(castle_solutions_2)) #create user column
+
+#gather the data to tidy form
+castle_solutions_2 <- gather(castle_solutions_2, key = "castle", value = "number_of_soldiers", 
+                             contains("castle_"), -strategy, -user)
+
+#reorder
+castle_solutions_2 <- castle_solutions_2[c("user", "castle", "number_of_soldiers", "strategy")]
```
Comment (beanumber) [inline data-raw/process_data_set_irene.R]: Please use `select()`

File: data-raw/process_data_set_irene.R [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+#classes of variables: only adjusted the user column, the rest are the appropriate class
+castle_solutions_2 <- castle_solutions_2%>%
```
Comment (beanumber) [inline data-raw/process_data_set_irene.R]: space

File: data-raw/process_data_set_irene.R [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  clean_names()
+
+castle_solutions_3$user <- seq.int(nrow(castle_solutions_3)) #create user column
+
+#gather the data to tidy form
+castle_solutions_3 <- gather(castle_solutions_3, key = "castle", value = "number_of_soldiers", 
+                             contains("castle_"), -strategy, -user)
+
+#reorder
+castle_solutions_3 <- castle_solutions_3[c("user", "castle", "number_of_soldiers", "strategy")]
```
Comment (beanumber) [inline data-raw/process_data_set_irene.R]: ditto

File: process_data_set_irene.R (hunk 2 above)
Comment (beanumber) [reply process_data_set_irene.R]: You have to delete this file and then commit the deletion to get it out of here.

**Review by beanumber — CHANGES_REQUESTED** [review]

**Review by rudeboybert — APPROVED** [review]

**Comment by rudeboybert** [comment]
Thanks @ireneryan, you're now listed as a package contributor: https://github.com/rudeboybert/fivethirtyeight/graphs/contributors

**Pull request merged**